text/0000-global-allocators.md

```diff
@@ -240,9 +252,47 @@ The allocator APIs could be simplified to a more "traditional"
 malloc/calloc/free API at the cost of an efficiency loss when using allocators
 with more powerful APIs.
 
+The global allocator could be an instance of the `Allocator` trait. Since that
+trait's methods take `&mut self`, things are a bit complicated however. The
+allocator would most likely need to be a `const` type implementing `Allocator`
+since it wouldn't be sound to interact with a static. This may cause confusion
```

Comment:
> I may be missing something here, but the issue is that you may not obtain a mutable reference to a static, but every method on Allocator takes &mut self

Ah, I see. So then, does the `Allocator` trait have to change? Or do we make the allocator unsafe and use `static mut`? If neither is possible, then we might need to switch back to the attributes approach or write a new trait with the hope of coalescing them some time...

> @mark-i-m There is no constructor.

Most allocators need some setup, though. Is the intent to just do something like `lazy_static`? That would annoy me, but it would work, I guess. Alternately, we could add a method to the interface to do this sort of set up...